### ISSUE TITLE
Implement shooting

### ISSUE THREAD

**Issue by MortenTobiasNielsen** [MortenTobiasNielsen/Factorio-TAS-Generator#41]
The tool won't really be complete before the user is able to shoot.

**Comment by theis999** [comment]
Zaspar has been talking about it. It would be hilarious, but it doesn't seem like a high priority.

**Comment by MortenTobiasNielsen** [comment]
Agreed. It is a nice to have when all the actual quality of life features are implemented. :)

**Comment by theis999** [comment]
There are two options: 
1. Shoot X times 
2. Shoot for X amount of time

2 is vastly simpler, as we don't have take into account how long it takes to fire a shot. On the other hand, the first option is more inline with the tool.
So i will likely implement 2. 

Either way, this would open up the tool to lack of placing armor, weapons and ammo in their corresponding slots.

**Comment by MortenTobiasNielsen** [comment]
I guess the user relatively fast will figure out how long it takes to fire a shot, so I think 2 is fine. 

That is true, but I highly doubt that a user will have a need to fire more than 10 magazines in a any% run and it seems like it could create a lot of changes to have to do the whole placing things, so I don't think that should be done before we do some kind of "you can now do a TAS with enemies" update - if that is ever going to be a thing. Enemy behavior would have to be deterministic for it to be remotely plausible to do that and I am not sure is the case.

**Comment by theis999** [comment]
I think enemies are deterministic the same way rock yields are, which is already annoying enough.

As long as any% is the longest run, then there isn't much need, no. But in 100% it could be useful to have the ability to use exoskeletons.

**Comment by MortenTobiasNielsen** [comment]
Aren't deterministic that it will always be the same? 

Agreed it would be nice to be able to equip items for longer runs, but you would have to be a special kind of crazy to want to attempt a 100% TAS. :D
